### ISSUE TITLE
../gedit-3.22.0/configure: line 18832: syntax error near unexpected token `maximum'

### ISSUE THREAD

**Issue by foulow** [Alexpux/MINGW-packages#3212]
Hi, I was compiling GTK+ on a Windows 10 x32 system. and I got this error on MSYS2 when using this command `makepkg-mingw -sL`, and this is the output I got from the shell:

```
1. ==> Starting build()...
2. configure: loading site script /mingw32/etc/config.site
3. checking for a BSD-compatible install... /usr/bin/install -c
4. checking whether build environment is sane... yes
5. checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
6. checking for gawk... gawk
7. checking whether make sets $(MAKE)... yes
8. checking whether make supports nested variables... yes
9. checking whether UID '197608' is supported by ustar format... yes
10. checking whether GID '197608' is supported by ustar format... yes
11. checking how to create a ustar tar archive... gnutar
12. checking whether to enable maintainer-specific portions of Makefiles... yes
13. checking whether make supports nested variables... (cached) yes
14. checking for gcc... gcc
15. checking whether the C compiler works... yes
16. checking for C compiler default output file name... a.exe
17. checking for suffix of executables... .exe
18. checking whether we are cross compiling... no
19. checking for suffix of object files... o
20. checking whether we are using the GNU C compiler... yes
21. checking whether gcc accepts -g... yes
22. checking for gcc option to accept ISO C89... none needed
23. checking whether gcc understands -c and -o together... yes
24. checking for style of include used by make... GNU
25. checking dependency style of gcc... gcc3
26. checking whether make sets $(MAKE)... (cached) yes
27. checking for special C compiler options needed for large files... no
28. checking for _FILE_OFFSET_BITS value needed for large files... 64
29. checking for pkg-config... /mingw32/bin/pkg-config
30. checking pkg-config is at least version 0.9.0... yes
31. checking for gcc... gcc
32. checking whether we are using the GNU Objective C compiler... yes
33. checking whether gcc accepts -g... yes
34. checking dependency style of gcc... gcc3
35. checking build system type... i686-w64-mingw32
36. checking host system type... i686-w64-mingw32
37. checking how to print strings... printf
38. checking for a sed that does not truncate output... /usr/bin/sed
39. checking for grep that handles long lines and -e... /usr/bin/grep
40. checking for egrep... /usr/bin/grep -E
41. checking for fgrep... /usr/bin/grep -F
42. checking for ld used by gcc... C:/msys32/mingw32/i686-w64-mingw32/bin/ld.exe
43. checking if the linker (C:/msys32/mingw32/i686-w64-mingw32/bin/ld.exe) is GNU ld                                                                                                                ... yes
44. checking for BSD- or MS-compatible name lister (nm)... /mingw32/bin/nm -B
45. checking the name lister (/mingw32/bin/nm -B) interface... BSD nm
46. checking whether ln -s works... no, using cp -pR
47. checking the maximum length of command line arguments... 8192checking how to convert i686-w64-mingw32 file names to i686-w64-mingw32 format..                                                                                                                . func_convert_file_msys_to_w32
48. checking how to convert i686-w64-mingw32 file names to toolchain format... func_                                                                                                                convert_file_msys_to_w32
49. checking for C:/msys32/mingw32/i686-w64-mingw32/bin/ld.exe option to reload object files... -r
50. checking for objdump... objdump
51. checking how to recognize dependent libraries... file_magic ^x86 archive import|                                                                                                                ^x86 DLL
52. checking for dlltool... dlltool
53. checking how to associate runtime and link libraries... func_cygming_dll_for_imp                                                                                                                lib
54. checking for ar... ar
55. checking for archiver @FILE support... @
56. checking for strip... strip
57. checking for ranlib... ranlib
58. checking command to parse /mingw32/bin/nm -B output from gcc object... ok
59. checking for sysroot... no
60. checking for a working dd... /usr/bin/dd
61. checking how to truncate binary pipes... /usr/bin/dd bs=4096 count=1
62. checking for mt... no
63. checking if : is a manifest tool... no
64. checking how to run the C preprocessor... gcc -E
65. checking for ANSI C header files... yes
66. checking for sys/types.h... yes
67. checking for sys/stat.h... yes
68. checking for stdlib.h... yes
69. checking for string.h... yes
70. checking for memory.h... yes
71. checking for strings.h... yes
72. checking for inttypes.h... yes
73. checking for stdint.h... yes
74. checking for unistd.h... yes
75. checking for dlfcn.h... no
76. checking for objdir... .libs
77. checking if gcc supports -fno-rtti -fno-exceptions... no
78. checking for gcc option to produce PIC... -DDLL_EXPORT -DPIC
79. checking if gcc PIC flag -DDLL_EXPORT -DPIC works... yes
80. checking if gcc static flag -static works... yes
81. checking if gcc supports -c -o file.o... yes
82. checking if gcc supports -c -o file.o... (cached) yes
83. checking whether the gcc linker (C:/msys32/mingw32/i686-w64-mingw32/bin/ld.exe)                                                                                                                 supports shared libraries... yes
84. checking whether -lc should be explicitly linked in... yes
85. checking dynamic linker characteristics... Win32 ld.exe
86. checking how to hardcode library paths into programs... immediate
87. checking whether stripping libraries is possible... yes
88. checking if libtool supports shared libraries... yes
89. checking whether to build shared libraries... yes
90. checking whether to build static libraries... no
91. checking for gtk-doc... no
92. configure: WARNING:
93.   You will not be able to create source packages with 'make dist'
94.   because gtk-doc >= 1.0 is not found.
95. checking for gtkdoc-check... no
96. checking for gtkdoc-check... no
97. checking for gtkdoc-rebase... no
98. checking for gtkdoc-mkpdf... no
99. checking whether to build gtk-doc documentation... no
100. checking for glib-2.0 >= 2.10.0 gobject-2.0  >= 2.10.0... yes
101. checking for Win32 platform... yes
102. checking for native Win32... checking for windres... windres
103. yes
104. checking for Mac OS X platform... no
105. checking for native Mac OS X... no
106. checking whether NLS is requested... yes
107. checking for intltool >= 0.50.1... 0.51.0 found
108. checking for intltool-update... /usr/bin/intltool-update
109. checking for intltool-merge... /usr/bin/intltool-merge
110. checking for intltool-extract... /usr/bin/intltool-extract
111. checking for xgettext... /mingw32/bin/xgettext
112. checking for msgmerge... /mingw32/bin/msgmerge
113. checking for msgfmt... /mingw32/bin/msgfmt
114. checking for gmsgfmt... /mingw32/bin/msgfmt
115. checking for perl... /usr/bin/perl
116. checking for perl >= 5.8.1... 5.24.1
117. checking for XML::Parser... ok
118. checking for msgfmt... (cached) /mingw32/bin/msgfmt
119. checking for gmsgfmt... (cached) /mingw32/bin/msgfmt
120. checking for xgettext... (cached) /mingw32/bin/xgettext
121. checking for msgmerge... (cached) /mingw32/bin/msgmerge
122. checking for ld used by GCC... C:/msys32/mingw32/i686-w64-mingw32/bin/ld.exe
123. checking if the linker (C:/msys32/mingw32/i686-w64-mingw32/bin/ld.exe) is GNU ld                                                                                                                ... yes
124. checking for shared library run path origin... done
125. checking for CFPreferencesCopyAppValue... no
126. checking for CFLocaleCopyCurrent... no
127. checking for GNU gettext in libc... no
128. checking for iconv... yes
129. checking for working iconv... yes
130. checking how to link with libiconv... /mingw32/lib/libiconv.dll.a -L/mingw32/lib
131. checking for GNU gettext in libintl... yes
132. checking whether to use NLS... yes
133. checking where the gettext function comes from... external libintl
134. checking how to link with libintl... /mingw32/lib/libintl.dll.a -L/mingw32/lib
135. checking for            libsoup-2.4... yes
136. checking for gspell-1 >= 0.2.5... yes
137. checking whether /mingw32/bin/python3 version is >= 3.2.3... yes
138. checking for /mingw32/bin/python3 version... 3.6
139. checking for /mingw32/bin/python3 platform... win32
140. checking for /mingw32/bin/python3 script directory... ${prefix}/lib/python3.6/si                                                                                                                te-packages
141. checking for /mingw32/bin/python3 extension module directory... ${exec_prefix}/l                                                                                                                ib/python3.6/site-packages
142. checking for pygobject-3.0 >= 3.0.0... yes
143. checking for
144.         libxml-2.0 >= 2.5.0
145.         glib-2.0 >= 2.44
146.         gio-2.0 >= 2.44
147.         gmodule-2.0
148.         gtk+-3.0 >= 3.21.3
149.         gtksourceview-3.0 >= 3.21.2
150.         libpeas-1.0 >= 1.14.1
151.         libpeas-gtk-1.0 >= 1.14.1
152.         gsettings-desktop-schemas
153. ... yes
154. checking whether to forbid deprecated symbols... no
155. checking for itstool... itstool
156. checking for xmllint... xmllint
157. checking for gobject-introspection... yes
158. checking for cos in -lm... yes
159. checking for  gtk+-3.0 >= 3.7.10 ... yes
160. checking for pkg-config... (cached) /mingw32/bin/pkg-config
161. checking pkg-config is at least version 0.16... yes
162. checking for pkg-config... (cached) /mingw32/bin/pkg-config
163. checking pkg-config is at least version 0.9.0... yes
164. checking for vapigen... yes
165. checking for glib-compile-resources... /mingw32/bin/glib-compile-resources
166. checking for glib-mkenums... /mingw32/bin/glib-mkenums
167. ../gedit-3.22.0/configure: line 18832: syntax error near unexpected token `maximum'
168. ../gedit-3.22.0/configure: line 18832: `GNOME_COMPILE_WARNINGS(maximum)'
169. ==> ERROR: A failure occurred in build().
170.     Aborting...
171. 
```

I hope someone can help me with this.

**Comment by foulow** [comment]
do I have to edit the file on ../gedit-3.22.0/configure? and what should I fix on line 18832 so it can be compiled?

**Comment by lazka** [comment]
Try `pacman -S mingw-w64-i686-gnome-common`

It's likely missing a build dependency

**Comment by foulow** [comment]
This time it went further, but now, I get this error:

```
1. checking whether /mingw32/bin/python3 version is >= 3.2.3... yes
2. checking for /mingw32/bin/python3 version... 3.6
3. checking for /mingw32/bin/python3 platform... win32
4. checking for /mingw32/bin/python3 script directory... ${prefix}/lib/python3.6/site-packages
5. checking for /mingw32/bin/python3 extension module directory... ${exec_prefix}/lib/python3.6/site
    -packages
6. checking for pygobject-3.0 >= 3.0.0... yes
7. checking for
8.         libxml-2.0 >= 2.5.0
9.         glib-2.0 >= 2.44
10.         gio-2.0 >= 2.44
11.         gmodule-2.0
12.         gtk+-3.0 >= 3.21.3
13.         gtksourceview-3.0 >= 3.21.2
14.         libpeas-1.0 >= 1.14.1
15.         libpeas-gtk-1.0 >= 1.14.1
16.         gsettings-desktop-schemas
17. ... yes
18. checking for
19.                 x11
20.         ... no
21. configure: error: Package requirements (
22.                 x11
23.         ) were not met:
24. 
25. No package 'x11' found
26. 
27. Consider adjusting the PKG_CONFIG_PATH environment variable if you
28. installed software in a non-standard prefix.
29. 
30. Alternatively, you may set the environment variables X11_CFLAGS
31. and X11_LIBS to avoid the need to call pkg-config.
32. See the pkg-config man page for more details.
33. ==> ERROR: A failure occurred in build().
34.     Aborting...
```
I am following the instructions from: [https://www.gtk.org/download/windows.php](https://www.gtk.org/download/windows.php) and I don't know where to put the PKGBUILD and gedit-i686.install files on this directory **C:\msys32\tmp\build_dir** I created.

**Comment by lazka** [comment]
Pushed some fixes in #3215
